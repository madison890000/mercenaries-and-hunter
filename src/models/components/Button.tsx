import {LoadingButton} from "@mui/lab";
import React, {useEffect, useState} from "react";


const Button = ({onClick, loading: outerLoading, children}: any) => {
    const [loading, setLoading] = useState(false);
    useEffect(() => {
        setLoading(outerLoading)
    }, [outerLoading])
    return (
        <LoadingButton loading={loading} onClick={async () => {
            setLoading(true);
            try {
                await onClick?.();
            } finally {
                setLoading(false);
            }
        }}>
            {children}
        </LoadingButton>
    )
}

export default Button