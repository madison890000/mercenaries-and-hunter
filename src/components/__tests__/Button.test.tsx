import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';

test('renders Button', () => {
    const DividerDom = renderer.create(
        <Button>
            Test Button
        </Button>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});
