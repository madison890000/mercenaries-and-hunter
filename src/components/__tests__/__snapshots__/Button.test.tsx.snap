// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1vqlznj-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
  disabled={false}
  id=":r0:"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Test Button
</button>
`;
